pageserver/src/pgdatadir_mapping.rs

```diff
@@ -1183,6 +1195,29 @@ impl<'a> DatadirModification<'a> {
         Ok(())
     }
 
+    pub(crate) fn put_rel_page_image_zero(&mut self, rel: RelTag, blknum: BlockNumber) {
+        self.pending_zero_data_pages
+            .insert(rel_block_to_key(rel, blknum).to_compact());
```

Comment:
Why not count these zero pages towards `pending_bytes`?

Reply:
Hmm, this was an oversight: we can account for this properly if we add here and then subtract if we later remove the zero page: 562b3706e

(While writing this I also have quite a strong urge to reprezent zero pages in a more compact way than literally writing 8192 bytes of zeros...)